### PR TITLE
Re-engineer build

### PR DESCRIPTION
**Pull request by gusty**
This will use standard `dotnet` commands to build, test and package the project.

For the doc generation, a separate .fsx script is provided. Must be run with `fsi`. See contents of the .fsx for instructions on how to use it..

### CODE REVIEW

File: buildDocs.fsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
+// Generate and Release Docs
```
Comment (wallymathieu) [inline buildDocs.fsx]: Neat!

File: tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <FscToolExe>$(FSC_ExePathCompilerBuild)</FscToolExe>
   </PropertyGroup>
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net452</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net45</TargetFrameworks>
```
Comment (wallymathieu) [inline tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj]: net45, do you want net47 here?

Comment (smoothdeveloper) [reply tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj]: For a library, better to target lowest possible framework I think.

FSharp.Core targets net45.

Comment (wallymathieu) [reply tests/FSharpPlus.Tests/FSharpPlus.Tests.fsproj]: Yes, so are there benefits of testing on net45?

File: Directory.Build.props [hunk 3]
```diff
@@ -1,4 +1,23 @@
-<Project>
+<Project ToolsVersion="15.0">
+
+  <PropertyGroup>
+    <Authors>gusty</Authors>
+    <PackageProjectUrl>https://github.com/jet/XRay/</PackageProjectUrl>
+    <PackageTags>tracing;fsharp</PackageTags>
+    <RepositoryType>git</RepositoryType>
+    <RepositoryUrl>https://github.com/jet/XRay.git</RepositoryUrl>
```
[the remark this answers is not in the document]
Comment (gusty) [reply Directory.Build.props]: and tags ;)

[the remark this answers is not in the document]
Comment (wallymathieu) [reply Directory.Build.props]: Indeed 😄

**Comment by wallymathieu** [comment]
For some reason the tests are not building. Seems there is some issue with locating system runtime. Perhaps there is a patch fix to the .net core 3 version since that's what seems to have changed on Mac and Linux.

**Comment by wallymathieu** [comment]
I was wrong, the fix was to bump the test project to net452

**Pull request merged**